src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -599,171 +607,57 @@ namespace cryptonote
       // moneromooo ... only because I <3 him. 
       std::vector<uint64_t> need_tx_indices;
       need_tx_indices.reserve(new_block.tx_hashes.size());
-        
-      transaction tx;
-      crypto::hash tx_hash;
 
-      for(auto& tx_blob: arg.b.txs)
+      // Collect:
+      //   1) list of transaction blobs for txids in this block which are known (have_tx)
+      //   2) list of transaction indices in this block where its hash is yet unkown (need_tx_indices)
+      // We also remove tx entrys from extra_block_txs which we already knew about before this notification
+      for(size_t tx_idx = 0; tx_idx < new_block.tx_hashes.size(); ++tx_idx)
       {
-        if(parse_and_validate_tx_from_blob(tx_blob.blob, tx))
+        const crypto::hash &tx_hash = new_block.tx_hashes[tx_idx];
+
+        blobdata tx_blob;
+        std::vector<blobdata> tx_blobs;
```

Comment:
Small nitpick on performance, you can move `tx_blobs` and `missed_txs` before the loop, and `.clear()` right here.

Reply:
This is a readability thing for me, but I personally don't like making the scope of variables any wider than it needs to be, especially for such an already complex function. If there's evidence that it measurably impacts performance, however, I would definitely be okay changing it to what you're suggesting.

[relevant link](https://wiki.sei.cmu.edu/confluence/display/c/DCL19-C.+Minimize+the+scope+of+variables+and+functions)